accepted/00104-enhanced-ansible-support.md

```diff
@@ -0,0 +1,296 @@
+- Feature Name: Enhanced Ansible Support
+- Start Date: 2024-12-04
+
+# Summary
+[summary]: #summary
+
+Uyuni provides support for Ansible, although on a very limited scale. The plan is to enhance the existing integration with some key features. 
+
+# Motivation
+[motivation]: #motivation
+
+Users wish to have Ansible more firmly integrated in Uyuni. We currently support the scheduling of Ansible playbooks through control nodes managed by Uyuni.
+
+We identified the following improvements to enhance the current implementation:
+
+1) Show the raw output results of a playbook in the event history of the control node along with a list of targeted inventory systems.
+2) Add UI support to allow editing of variables defined in the playbook.
+3) Support the recurrent scheduling of Ansible playbooks.
+4) Set default playbook and inventory paths when adding a new control node in Uyuni.
+5) Add option to filter by Ansible managed systems onboarded in Uyuni in the system list.
+
+These points will be discussed one-by-one in detail in the next section.
+
+# Detailed design
+[design]: #detailed-design
+
+## Show the raw output results of a playbook in the event history of the control node along with a list of targeted inventory systems.
+
+#### Raw output
+
+To show the raw output returned by the ansible command itself instead of the output returned by salt a change to the [runplaybook.sls](https://github.com/uyuni-project/uyuni/blob/master/susemanager-utils/susemanager-sls/salt/ansible/runplaybook.sls) will have to be made.
+
+We're currently using the `state module`:
+```
+run_ansible_playbook:
+  ansible.playbooks:
+     - .......
+```
+
+To get the raw output we will have to use module.run to call the `ansible.playbooks` function from the `execution module`:
+```
+run_ansible_playbook:
+  module.run:
+    - name: ansible.playbooks
+    - ....
+```
+
+The changed `runplaybook.sls` would then look like:
+```
+run_ansible_playbook:
+  mgrcompat.module_run:
+    - name: ansible.playbooks
+    - playbook: {{ pillar["playbook_path"] }}
+    - rundir: {{ pillar["rundir"] }}
+    - flush_cache: {{ pillar["flush_cache"] }}
+{%- if "inventory_path" in pillar %}
+    - inventory: {{ pillar["inventory_path"] }}
+{% endif %}
+```
+
+#### Displaying targeted systems
+
+Since the list of systems targeted by an ansible playbook can be very long, I propose to display the used inventory as a link in the event history details instead.
+
+In case the selected event details are of the playbook action type we will set an additional request parameter called `inventory` in `SystemHistoryEventAction.java` and render an extra row that would look like:
+```
+Inventory: <link_to_inventory>
+```
+
+Clicking the link would then forward the user to the inventory file in `System` > `Ansible` > `Inventories`.
```

Comment:
It's a tricky one. So far, we haven't been keeping track of any systems of inventory. Yes, currently, if inventory is removed from the control node, we will have an orphan inventory link in our database IIRC.

I would suggest to just make this targeted systems part of playbook action by parsing the output or inventory at schedule time. Keeping it linked with inventory will create the issue what you just brought up, Ricardo. Yes, it's understood that list could be long.

Reply:
Abid I think that is a good option.
A possible alternative is to save the inventory link/name at schedule time, and then at execution time we save the file content in the database, for example as part of the execution result. 
I'm not sure what would be better, but one way or the other we should save the inventory list without link to the file.

The main good point of just saving in the result is for recurrent executions if users change the inventory file it will reflect be reflected in the next recurrent execution, and as user I would expect that to happen.

Reply:
I like option 2 more, it's more clean and direct.